core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -97,16 +99,55 @@
       new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
           RelFactories.LOGICAL_BUILDER);
 
+  private EnumSet<SqlKind> functionsToReduce;
+
   //~ Constructors -----------------------------------------------------------
 
   /** Creates an AggregateReduceFunctionsRule. */
   public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
       RelBuilderFactory relBuilderFactory) {
     super(operand, relBuilderFactory, null);
+    // by default, this rule will reduce all functions it handles
+    functionsToReduce = EnumSet.noneOf(SqlKind.class);
+    addDefaultSetOfFunctionsToReduce();
+  }
+
+  /**
+   * Creates an AggregateReduceFunctionsRule with client
```

Comment:
Update also the javadoc of `reduceAggs` in line 186 according to the new behavior.

Reply:
done